jupyterhub/apihandlers/users.py

```diff
@@ -384,7 +394,14 @@ async def post(self, user_name):
                     "Error authenticating request for %s: %s", self.request.uri, e
                 )
                 raise web.HTTPError(403)
+            if name is None:
+                raise web.HTTPError(403)
             requester = self.find_user(name)
+        else:
+            # perform delayed xsrf check
+            # if we aren't authenticating via the request body
+            self._skip_post_check_xsrf = False
+            self.check_xsrf_cookie()
         if requester is None:
             # couldn't identify requester
             raise web.HTTPError(403)
```

Comment:
It think it could make sense to re-locate the if statement on line 388-390 within the initial if block starting under 369, not important though.

Reply:
That would make sense and definitely still be correct. I like having it here, even though it can result in one redundant `if x is None` check because it's an unconditional block (if nobody identified, definitely stop), rather than having more complex branches. It helps me be more confident that the handler will not proceed past this point if auth has failed, when auth can be achieved via a few different paths.